server/shared/procBuilder.go

```diff
@@ -87,6 +87,16 @@ func (b *ProcBuilder) Build() ([]*BuildItem, error) {
 			metadata := metadataFromStruct(b.Repo, b.Curr, b.Last, proc, b.Link)
 			environ := b.environmentVariables(metadata, axis)
 
+			// add GLOBAL_ environment variables for substituting
+			for k, v := range b.Envs {
+				new_k := "GLOBAL_" + k
```

Comment:
Couldn't we swap the global env vars with `b.environmentVariables(metadata, axis)` in line 88. That way the global value can be overwritten, but the user wouldn't have to remember to prefix his vars with `GLOBAL_`.

Reply:
I'm not sure I follow. `b.eV(m,a)` provides the `CI_` variables (which probably shouldn't be overwritten). I can easily remove the prefix and either just skip anything that starts with `CI_` or not check and allow them to be overwritten. Despite the annoyance of having to remember the prefix, it seemed like a good safety measure to avoid overwriting the builtin variables.
Am I misunderstanding your suggestion?

Reply:
I'm fine with this as is